pkg/backend/filestate/backend_test.go

```diff
@@ -395,3 +395,12 @@ func TestRenameWorks(t *testing.T) {
 	assert.Len(t, history, 1)
 	assert.Equal(t, apitype.DestroyUpdate, history[0].Kind)
 }
+
+func TestLoginToNonExistingFolderFails(t *testing.T) {
```

Comment:
Does this test actually test the new behavior? I'm unsure from your PR comments.

Reply:
No, it doesn't. Only remote file backends are currently problematic - but I wasn't sure if we have any prior art of testing against the S3 backend specifically - I think I wasn't able to spot any.

Reply:
I'm worried this might be showing that we've regressed here? Did this used to work before the IsAccessible call? Might be gocloud would of just mkdir'd the folder when needed?

You've manually tested this with an s3 bucket right? I'm not overly worried about getting an s3 test in place just for this, although we should probably have some sort of integration test for it added at some point.

Reply:
> I'm worried this might be showing that we've regressed here? 

I'm sorry, I haven't caught this comment. "this" what and where have we regressed? This test is green as far as I can tell, and (somewhat unfortunately) it would be green without my change.

> You've manually tested this with an s3 bucket right?

Yes, I did. Logging into an existing S3 bucket succeeds and logging into a random guid bucket fails.